pkg/cmd/util/clientcmd/factory.go

```diff
@@ -512,6 +518,20 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	return w
 }
 
+// useDiscoveryRESTMapper checks the server version to see if its recent enough to have
+// enough discovery information avaiable to reliably build a RESTMapper.  If not, use the
+// hardcoded mapper in this client (legacy behavior)
+func useDiscoveryRESTMapper(serverVersion string) bool {
+	serverSemVer, err := semver.Parse(serverVersion[1:])
```

Comment:
> how does this handle emptystring (or whatever values ServerVersion() can return for legacy servers)?

We never shipped a level that gave back empty string.  If there was an error getting the version, we handle that before this call.